internal/devconfig/packages.go

```diff
@@ -154,6 +150,19 @@ func NewPackage(name string, values map[string]any) Package {
 	}
 }
 
+func (p *Package) VersionedName() string {
+	name := p.name
+	if p.Version != "" {
+		name += "@" + p.Version
+	}
+	return name
+}
+
+func (p *Package) IsEnabledOnPlatform() (bool, error) {
```

Comment:
Making this return `(bool, error)` causes large changes in the code base (and generally makes things  bit more difficult to read). 

If seems like in the follow up PR you implement this using `nix.System`

Suggestion:

Have `nix.EnsureNixInstalled` call `nix.ComputeSystem` which stores the system in a package variable.

nix.System returns it (and panics if not set).

I know this has a bunch of stuff stacked on it so I don't know how easy it would be to make this change, but it would prevent a good amount of code/blame churn.

Reply:
I added `nix.MustGetSystem`, since `nix.System` currently has an error return refactoring which would introduce a lot of unrelated changes. Can cleanup in a followup PR.